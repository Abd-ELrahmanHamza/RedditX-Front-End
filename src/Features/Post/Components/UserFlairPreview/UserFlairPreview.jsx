import profile from "Features/Post/Assets/Images/Post-guidelines.png";
import { useState, useEffect } from "react";
import { HiOutlinePencil } from "react-icons/hi";
import FlairModal from "../FlairModal/FlairModal";
import {
  Container,
  DataContainer,
  Flair,
  FlairContainer,
  FlairPreview,
  Name,
  NameAndFlairContainer,
  PreviewContainer,
  ProfilePic,
  ProfilePicContainer,
  UserFlairContainer,
} from "./UserFlairPreview.styled";

// API services
import getPostFlairs from "Features/Post/Services/getFlairs";
import useFetchFunction from "Hooks/useFetchFunction";
import { useAuth } from "Features/Authentication/Contexts/Authentication";

/**
 * component that preview your subreddit flair
 *
 * @returns {React.Component} CreatePost component
 */
const UserFlairPreview = () => {
  //state to show modal
  const [modalShow, setModalShow] = useState(false);

  // State for selected flair index
  const [flairIndex, setFlairIndex] = useState(null);
  /**
   * handler on click on pen handler
   */
  function clickPenHandler() {
    setModalShow(true);
  }

  // const [flairs, error, isLoading, reload] = useFetch({
  //   axiosInstance: axios,
  //   method: "GET",
  //   url: "/flairs/",
  //   requestConfig: {
  //     headers: {
  //       "Content-Language": "en-US",
  //     },
  //   },
  // });
  // Fetch flairs
  const auth = useAuth();
  const [flairs, error, isLoading, fetchData] = useFetchFunction();
  useEffect(() => {
    getPostFlairs(fetchData, auth);
  }, []);

  /**
   *
   * @returns {React.Component} CreatePost component
   */
  const Preview = () => {
    return (
      <PreviewContainer>
        <FlairPreview>
          user flair preview
          <button onClick={clickPenHandler}>
            <span className="icon">
              <HiOutlinePencil />
            </span>
          </button>
        </FlairPreview>
        <UserFlairContainer>
          <DataContainer>
            <ProfilePicContainer>
              <ProfilePic profile={profile} />
            </ProfilePicContainer>
            <NameAndFlairContainer>
              <Name>
                <span>user name</span>
              </Name>
              <FlairContainer>
                {flairIndex !== null && (
                  <Flair
                    style={{
                      color: flairs[flairIndex].flairTextColor,
                      backgroundColor: flairs[flairIndex].flairBackGroundColor,
                    }}
                  >
                    {flairs[flairIndex].text}
                  </Flair>
                )}
              </FlairContainer>
            </NameAndFlairContainer>
          </DataContainer>
        </UserFlairContainer>
      </PreviewContainer>
    );
  };

  return (
    <Container>
      {/* dynamic display cuz if there is no display */}
      <Preview />
      <FlairModal
        show={modalShow}
        onHide={() => setModalShow(false)}
        flairList={flairs}
        flairIndex={flairIndex}
        setFlairIndex={setFlairIndex}
        error={error}
        isLoading={isLoading}
        postOrUser={"user"}
      />
    </Container>
  );
};

export default UserFlairPreview;
