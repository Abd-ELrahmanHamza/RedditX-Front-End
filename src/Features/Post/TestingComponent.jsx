// Import contexts providers
import { SubmitDestinationProvider } from "Features/Post/Contexts/submitDestination";
import { CreatePostTitleProvider } from "Features/Post/Contexts/createPostTitle";
import { CreatePostFlagsProvider } from "Features/Post/Contexts/createPostFlags";
import { CreatePostAttachmentsProvider } from "Features/Post/Contexts/createPostAttachments";
import { CreatePostFlairsProvider } from "Features/Post/Contexts/createPostFlairs";
import { CreatePostTextProvider } from "Features/Post/Contexts/createPostText";
import {
  AuthProvider,
  useAuth,
} from "Features/Authentication/Contexts/Authentication";

// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";

// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { useEffect } from "react";

const AuthTesting = ({ children }) => {
  const auth = useAuth();
  useEffect(() => {
    auth.login({
      expiresIn: 86400,
      token:
        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbFR5cGUiOiJiYXJlIGVtYWlsIiwidXNlcm5hbWUiOiJ0Ml9ob21lIiwiaWF0IjoxNjY5OTI0MDEzLCJleHAiOjE2NzAzNTYwMTN9.FDa59vJi5gWy4dQVkEtgyCfm3Pp2BtHkPz0_AfNVKgY",
      username: "t2_home",
    });
  }, []);

  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        <SubmitDestinationProvider>
          <CreatePostTitleProvider>
            <CreatePostFlagsProvider>
              <CreatePostAttachmentsProvider>
                <CreatePostFlairsProvider>
                  <CreatePostTextProvider>{children}</CreatePostTextProvider>
                </CreatePostFlairsProvider>
              </CreatePostAttachmentsProvider>
            </CreatePostFlagsProvider>
          </CreatePostTitleProvider>
        </SubmitDestinationProvider>
      </ThemeProvider>
    </BrowserRouter>
  );
};

const TestingComponent = ({ children }) => {
  return (
    <AuthProvider>
      <AuthTesting>{children}</AuthTesting>
    </AuthProvider>
  );
};

export default TestingComponent;
