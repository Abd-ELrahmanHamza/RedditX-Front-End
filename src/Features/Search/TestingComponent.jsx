// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";
// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { SafeContextProvider } from "./Contexts/SafeSearchContext/Safe-context";
import { SearchContextProvider } from "./Contexts/SearchWordContext/Search-context";

const TestingComponent = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        <SearchContextProvider>
          <SafeContextProvider>{children}</SafeContextProvider>
        </SearchContextProvider>
      </ThemeProvider>
    </BrowserRouter>
  );
};

export default TestingComponent;
