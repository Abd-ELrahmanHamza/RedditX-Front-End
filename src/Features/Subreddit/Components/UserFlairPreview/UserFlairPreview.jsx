import profile from "Features/Subreddit/Assets/index.jpeg";
import { useIsBanned } from "Features/Subreddit/Contexts/IsBannedProvider";
import { useIsMuted } from "Features/Subreddit/Contexts/IsMutedProvider";
import { useState } from "react";
import { HiOutlinePencil } from "react-icons/hi";
import { Separator } from "../CommunityCard/CommunityCard.styled";
import SelectYourFlair from "../SelectYourFlair/SelectYourFlair";
import RequestMods from "../RequestMods/RequestMods";
import {
  Container,
  CreatePost,
  DataContainer,
  Flair,
  FlairContainer,
  FlairPreview,
  Name,
  NameAndFlairContainer,
  PreviewContainer,
  ProfilePic,
  ProfilePicContainer,
  RequestPostButton,
  UserFlairContainer,
} from "./UserFlairPreview.styled";

/**
 * component that preview your subreddit flair
 *
 * @returns {React.Component}
 */
const UserFlairPreview = () => {
  const [showModal, setShowModal] = useState(false);

  const { isBanned } = useIsBanned();
  const { isMuted } = useIsMuted();

  // const flair = { text: "", backgroundColor: "", color: "" };
  const [flair, setFlair] = useState({
    text: "hi",
    color: "white",
    backgroundColor: "#000",
  });
  //state to show modal
  const [modalShow, setModalShow] = useState(false);

  /**
   * handler on click on pen handler
   */
  function clickPenHandler() {
    setModalShow(true);
  }

  function ReqPostHandler() {
    setShowModal(true);
  }

  /**
   * handler when click save on flair modal that takes new flair selected
   *
   * @param {object} flair
   */
  function changeFlair(flair) {
    setModalShow(false);
    setFlair(flair);
  }

  /**
   * component preview your flair in this subreddit
   *
   * @returns {React.Component}
   */
  const Preview = () => {
    return (
      <PreviewContainer>
        <FlairPreview>
          user flair preview
          <button onClick={clickPenHandler}>
            <span className="icon">
              <HiOutlinePencil />
            </span>
          </button>
        </FlairPreview>
        <UserFlairContainer>
          <DataContainer>
            <ProfilePicContainer>
              <ProfilePic profile={profile} />
            </ProfilePicContainer>
            <NameAndFlairContainer>
              <Name>
                <span>user name</span>
              </Name>
              <FlairContainer>
                <Flair
                  style={{
                    color: flair.color,
                    backgroundColor: flair.backgroundColor,
                  }}
                >
                  {flair.text}
                </Flair>
              </FlairContainer>
            </NameAndFlairContainer>
          </DataContainer>
        </UserFlairContainer>
      </PreviewContainer>
    );
  };

  return (
    <Container>
      {!isBanned && !isMuted && (
        <CreatePost to="/submit/" className="create-post">
          Create Post
        </CreatePost>
      )}
      {!isBanned && isMuted && (
        <RequestPostButton onClick={ReqPostHandler}>
          Request to Post
        </RequestPostButton>
      )}
      {/* dynamic display cuz if there is no display */}
      {true && <Separator />}
      {true && <Preview />}
      <SelectYourFlair
        modalShow={modalShow}
        hide={() => setModalShow(false)}
        onApply={changeFlair}
      />
      <RequestMods
        modalShow={showModal}
        hide={() => setShowModal(false)}
        isPriv={false}
      />
    </Container>
  );
};

export default UserFlairPreview;
