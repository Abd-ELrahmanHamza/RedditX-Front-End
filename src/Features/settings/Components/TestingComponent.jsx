import {
  AuthProvider,
  useAuth,
} from "Features/Authentication/Contexts/Authentication";

// Import react router dom
import { BrowserRouter, Route, Routes } from "react-router-dom";

// Import themes
import darkTheme from "Theme/darkTheme";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";

// Import theme provider from styled components
import { ThemeProvider } from "styled-components";
import { useEffect } from "react";

const AuthTesting = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={{ ...defaultTheme, ...lightTheme }}>
        {children}
      </ThemeProvider>
    </BrowserRouter>
  );
};

const TestingComponent = ({ children }) => {
  return (
    <AuthProvider>
      <AuthTesting>{children}</AuthTesting>
    </AuthProvider>
  );
};

export default TestingComponent;
