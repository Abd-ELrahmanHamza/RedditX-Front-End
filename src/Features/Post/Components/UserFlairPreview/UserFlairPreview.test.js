import { render } from "@testing-library/react";
import Enzyme, { shallow } from "enzyme";
import TestingComponent from "Features/Post/TestingComponent";
import UserFlairPreview from "./UserFlairPreview";

jest.mock("Features/Post/Services/getFlairs", () => (fetchData, auth) => {});

jest.mock("Hooks/useFetchFunction.js", () => () => {
  return [
    "data",
    "error",
    "loading",
    jest.fn().mockReturnValue({
      data: {
        name: "test",
        type: "subreddit",
      },
    }),
  ];
});

jest.mock("Features/Authentication/Contexts/Authentication", () => ({
  __esModule: true, // this property makes it work
  useAuth: function () {
    return {
      getUserName: jest.fn().mockReturnValue("username"),
      getToken: jest.fn().mockReturnValue("token"),
      isLoggedIn: jest.fn().mockReturnValue(true),
    };
  },
  AuthProvider: ({ children }) => {
    return (
      <mock-authprovider data-testid="auth-provider">
        {children}
      </mock-authprovider>
    );
  },
}));

describe("UserFlairPreview", () => {
  it("should be able to render without crashing", async () => {
    render(
      <TestingComponent>
        <UserFlairPreview />
      </TestingComponent>
    );
  });
});
