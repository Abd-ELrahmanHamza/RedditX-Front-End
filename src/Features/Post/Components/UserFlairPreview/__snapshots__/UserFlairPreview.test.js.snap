// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preview of selected flair should render without crashing 1`] = `
<styled.div>
  <Preview />
  <FlairModal
    error=""
    flairIndex={null}
    flairList={
      Array [
        Object {
          "allowUserEdits": true,
          "flairBackGroundColor": "rgb(70, 209, 96)",
          "flairTextColor": "rgb(255, 255, 255)",
          "id": "t7_63248d012f459a937e2684fd",
          "modOnly": true,
          "text": "Flair 1 text",
        },
        Object {
          "allowUserEdits": true,
          "flairBackGroundColor": "blue",
          "flairTextColor": "rgb(255, 255, 255)",
          "id": "t7_63248d012f459a937e1223123d",
          "modOnly": true,
          "text": "Flair 2 text",
        },
        Object {
          "allowUserEdits": true,
          "flairBackGroundColor": "red",
          "flairTextColor": "rgb(70, 209, 96)",
          "id": "t1_632012f459a937e1223123d",
          "modOnly": true,
          "text": "Flair 3 text",
        },
      ]
    }
    isLoading={false}
    onHide={[Function]}
    postOrUser="user"
    setFlairIndex={[Function]}
    show={false}
  />
</styled.div>
`;
